.github/workflows/unit-test-on-pull-request.yml

```
@@ -96,6 +96,30 @@ jobs:
             exit 1
           fi
 
+  check-rust-blobs:
+    name: Check for differences in the Rust binary blobs
+    runs-on: ubuntu-24.04
+    container: otel/opentelemetry-ebpf-profiler-dev:latest
+    defaults:
+      run:
+        shell: bash --login {0}
+    steps:
+      - name: Clone code
+        uses: actions/checkout@v4
+      - name: Hash Rust blobs
+        run: |
+          sha256sum target/*-unknown-linux-musl/release/libsymblib_capi.a > rust-blobs.hash
+      - name: Rebuild Rust blobs
+        run: |
+          rm -rf target/
+          make rust-components TARGET_ARCH=amd64
+          make rust-components TARGET_ARCH=arm64
+      - name: Check for differences
+        run: |
+          if ! sha256sum --check rust-blobs.hash; then
+            echo "Please rebuild and commit the updated Rust binary blobs."
+            exit 1
+          fi
```

This can be simplified and sped up, as we don't need the extra step to hash and verify the hashes. We can also let the user know exactly which blob(s) actually changed, instead of printing a generic message:

```suggestion
      - name: Check for differences
        run: |
          if ! git diff --exit-code --name-only; then
            echo "Binary blob(s) changed, please rebuild and commit the updated blobs."
            exit 1
          fi
```

We should do the same for the tracers.

---

Merged the eBPF and Rust check with https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/415/commits/6e6fed4a1b1bdb1f6a0fece677909a5f45024edf. Please let me know, if this works for you.

---

Thanks!